src/main/java/com/pagsestagio/movieapi/controller/FilmeController.java

```diff
@@ -1,85 +1,82 @@
 package com.pagsestagio.movieapi.controller;
 
+import com.pagsestagio.movieapi.controller.resposta.FilmeResposta;
+import com.pagsestagio.movieapi.controller.resposta.FilmeRespostaErroRetornaMensagem;
+import com.pagsestagio.movieapi.controller.resposta.FilmeRespostaSucessoRetornaFilme;
+import com.pagsestagio.movieapi.controller.resposta.FilmeRespostaSucessoRetornaLista;
 import com.pagsestagio.movieapi.model.*;
+import com.pagsestagio.movieapi.model.resultado.FilmeResultado;
+import com.pagsestagio.movieapi.model.resultado.FilmeResultadoRetornaFilme;
+import com.pagsestagio.movieapi.service.FilmeService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.HashMap;
 import java.util.Map;
 
+
 @RestController
 @RequestMapping("/filmes")
 public class FilmeController {
 
-    Map<Integer, String> nomesDeFilmesPorId = new HashMap<>();
+    private final FilmeService filmeService;
 
-    @GetMapping("/{id}")
-    public ResponseEntity<FilmeResposta> mensagemRequisicaoGet(@PathVariable Integer id){
+    @Autowired
+    public FilmeController(FilmeService filmeService){
+        this.filmeService = filmeService;
+    }
 
+    @GetMapping("/{id}")
+    public ResponseEntity<FilmeResposta> pegarFilmePeloId(@PathVariable Integer id){
```

Comment:
Aliás, essa maneira de tratar retornos para o cliente com if/else talvez não seja tão bacana assim. Cria-se muito código repetido e todos têm a mesma função verificar se o retorno do service foi válido ou não, ou seja, quebrou ou não.
Meio que nessa abordagem estamos fazendo assim:

```
se (retorno do meu service == retorno que eu queria ) {
     retorna status ok
} 
se não {
    retorna status bad
}
```

Isso vai se repetir por todos os seus endpoints dessa maneira.
Para resolver isso, talvez seja interessante a começar a trabalhar com exeções e seus handlers de maneira mais profunda. 

No caso Get, temos algumas coisas que o cliente pode fazer errado ao tentar consultar, como por exemplo, passar uma String no lugar do ID que é necessáriamente um Integer ou até mesmo passar um ID que não exista e consequentemente sua consulta não irá retornar.

Talvez esses dois cenários, ao em vez de você se preocupar diretamente com eles, as exceptions podem fazer esse trabalho pra você. Há um recurso que fica responsável por capturar exeções que você definir e retornar um valor padrão como resposta caso a exeção seja capturada. Por exemplo, nesse mesmo cenário do Get, caso o cliente passe um ID tal qual não exista o FIlme hoje em dia você retorna isso no else com um badrequest, mas talvez exista um jeito mais elegante de se fazer isso, olha só.

Essa é uma classe de exeção personalizada, tal qual deve ser usada para coisas que não foram achadas como o nome diz.
```java
class NotFoundException():Exception {
     private String message;
     constructor() {...}
     public String getMessage() {...}
}
```
Essa com certeza é a exeção nesse nosso cenário que iria extourar caso o filme não fosse achado pelo Id. Bom, agora que temos a exeção devemos captura-la com o recurso que comentei mais a cima - ExceptionHandler.
Basicamente esse ExceptionHandler é uma classe que consegue intermediar as exceções que acontecem na sua aplicação e executar um bloco de código dependendo da execção que acontecer.

```java
@ControllerAdvice
class GlobalExceptionHandler {
     
      @ExceptionHandler(NotFoundException.class)
      public ResponseEntity<NotFoundResponse> response(NotFoundException e) {
            return ResponseEntity.status(404).body(new NotFoundResponse(e.getMessage()))
      }

}
```

Basicamente nós pegamos e adicionando essa classe de GlobalExceptionHandler, anotada com o @ControllerAdvice que faz todo o intermédio entre suas exeções da app, e cada exeção que você for desenvolvendo durante o código e que deve ser de alguma maneira tratada antes de voltar para o cliente é interessante você ir compondo essa classe até ter suas exeções mapeadas e sem ter que tratalas em todo lugar que você as usa.

Caso queira entender um teco mais: https://vanderloureiro.medium.com/retorno-amig%C3%A1vel-de-exce%C3%A7%C3%B5es-no-spring-com-exceptionhandler-3b91a425875c

Reply:
essa parte do controllerAdvice eu não dei um toque para ele para não ser muito coisa de uma vez.
depois abrir uma outra task para ele.